tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -149,6 +149,7 @@ Status StorageManagerCanonical::array_create(
         "' already exists"));
   }
 
+  // #TODO: replace this mutex with tiledb::sm::object_mtx_ upon migration.
```

Comment:
Both mutex occurrence need to be migrated simultaneously in order to preserve behavior. That's going to entail including `object_mutex.h` as a header in this file.